File: ch2_labs/lab5_C#/TextLibrary/Symvoly.cs
using System.Collections.Generic;

namespace TextLibrary
{
    public class Symvoly : Ryadok
    {
        public Symvoly(string value) : base(value) { }

        public override int GetLength() => text.Length;

        public override void Modify()
        {
            var result = new List<char>();
            for (int i = 0; i < text.Length; i++)
            {
                if ((i + 1) % 2 == 0)
                    result.Add(text[i]);
            }
            text = new string(result.ToArray());
        }
    }
}

File: ch2_labs/lab5_C++/src/Symvoly.cpp
#include "Symvoly.h"

Symvoly::Symvoly(const std::string& value) : Ryadok(value) {}

int Symvoly::getLength() const {
    return text.length();
}

void Symvoly::modify() {
    std::string result;
    for (size_t i = 0; i < text.length(); ++i) {
        if ((i + 1) % 2 == 0)
            result += text[i];
    }
    text = result;
}
